exes/ec/inc/public/tfc/ec/devices/beckhoff/EL2xxx.hpp

```diff
@@ -1,23 +1,25 @@
 #pragma once
 
 namespace tfc::ec::devices::beckhoff {
-template <typename manager_client_type, size_t size, uint32_t pc>
-class el200x : public base {
+template <typename manager_client_type, size_t size, uint32_t pc, std::string_view const& name>
+class el2xxx : public base {
 public:
-  explicit el200x(boost::asio::io_context& ctx, manager_client_type& client, uint16_t slave_index) : base(slave_index) {
+  explicit el2xxx(boost::asio::io_context& ctx, manager_client_type& client, uint16_t slave_index) : base(slave_index) {
     for (size_t i = 0; i < size; i++) {
-      bool_receivers_.emplace_back(
-          std::make_unique<tfc::ipc::bool_slot>(ctx, client, fmt::format("EL200{}.{}.out.{}", size, slave_index, i),
-                                                std::bind_front(&el200x::set_output, this, i)));
+      bool_receivers_.emplace_back(std::make_unique<tfc::ipc::bool_slot>(
+          ctx, client, fmt::format("{}.{}.out.{}", name, slave_index, i), std::bind_front(&el2xxx::set_output, this, i)));
     }
   }
 
   static constexpr uint32_t product_code = pc;
   static constexpr uint32_t vendor_id = 0x2;
 
   void process_data(std::span<std::byte>, std::span<std::byte> output) noexcept final {
-    static_assert(size <= 8);
+    static_assert(size == 8 || size == 4 || size == 16, "Invalid size");
     output[0] = static_cast<std::byte>(output_states_.to_ulong() & 0xff);
+    if constexpr (size > 8) {
+      output[1] = static_cast<std::byte>(output_states_.to_ulong() >> 8);
+    }
   }
 
 private:
```

Comment:
Not related to this PR but, `set_output` caches the last state set, will it ever be theoretically possible to miss a zeromq message, high -> low -> high, if it is theoretically possible should we add logs or queue events.

Reply:
It is possible to miss output events, the ethercat deamon works on a process loop with some cycle time while our ipc mechanism is event based.

What kind of logging are you thinking about? where some state change is explicitly changed or logging every value change?